improved testing/formatting section (#36576)
ros/rosdistro: CONTRIBUTING.md

When submitting pull requests it is expected that they pass the unit tests for formatting.
The unit tests enforce alphabetization of elements and a consistent formatting to keep merging as clean as possible.

#### Testing

> To run the tests run ``nosetests`` (nose tests) in the root of the repository.

Reviewer: A separate section is a good idea but maybe some other intro line or about nose https://nose.readthedocs.io/en/latest/index.html? It would be better for uses to have the command after installing dependencies.

> Note: There's a [known issue](https://github.com/disqus/nose-unittest/issues/2) discovered [here](https://github.com/ros/rosdistro/issues/16336) that most tests won't run if you have the python package `nose-unitttest` installed.

Reviewer: ```suggestion
Note: There's a [known issue](https://github.com/disqus/nose-unittest/issues/2) discovered [here](https://github.com/ros/rosdistro/issues/16336) that most tests won't run if you have the python package `nose-unittest` installed.
```

Note: There's a [known issue](https://github.com/disqus/nose-unittest/issues/2) discovered [here](https://github.com/ros/rosdistro/issues/16336) that most tests won't run if you have the python package `nose-unitttest` installed.

The contained tests require several dependencies that can be installed either from the ROS repositories or via pip (list built based on the content of [.travis.yaml](https://github.com/ros/rosdistro/blob/master/.travis.yml)). If you want to run the tests before submitting, first install the dependencies. Using `pip` is recommended, e.g. by this command:

```bash
python3 -m pip install -r test/requirements.txt
```

> To run the tests run ``nosetests`` in the root of the repository.

Reviewer: This line made sense to me after installing dependencies, and after the list of installed modules.

> The following modules/packages are needed and will see install by avobe comment.

Reviewer 2: ```suggestion
The following modules/packages are needed and will see install by above comment.
```

Author: thanks, used it with a bit more of needed change.

> The following modules/packages are needed and will see install by the above command.

Reviewer: ```suggestion
The following modules/packages are needed and will be installed by the above command.
```

| unidiff        | python-unidiff (Zesty and higher) | unidiff        |
| yamllint       | yamllint                          | yamllint       |

#### Formatting

> There is further the tool ``rosdistro_reformat`` which will fix most formatting errors such as alphabetization and correct formatting. This tool takes an ``index`` parameter that is a required command line argument. With option ``-n`` a dry run is possible. 

Reviewer: ```suggestion
There is a tool ``rosdistro_reformat`` which will fix most formatting errors such as alphabetization and correct formatting. This tool takes an ``index`` parameter that is a required command line argument. With option ``-n`` a dry run is possible. 
```
